### PR TITLE
Create stupid_ideas.txt

### PR DESCRIPTION
**Pull request by openfnord**
1st stupid idea: integrate ACRN into coreboot so correboot boots securely into ACRN hypervisior as security OS

so this accidentially found its way into projectacrn. As I just wanted to add it to my fork (I collect some source that I might use later).
While thinking on how to remove it. I thought, maybe you should just read my idea,
discuss it and reject my pull request :-)

Cheers

luja

### CODE REVIEW

File: stupid_ideas.txt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+Maybe this is helpful for me or others some point in time later.
+
+ACRN-Hypervisor:
+ACRN is booted using EFI and GRUB, this is complicated.
+So as ACRN wants to be embedded, so combine it with coreboot and boot into the hypervisor right away.
+Having a running ACRN hypervisor just after board bring up, booted from the flash of the x86_64 mainboard's 
+BIOS flash gives you a chain of trust while booting. 
+In the hypervisor ACRN you can try to boot something like linux, xBSD or commecrial operating systems.
+As ACRN claims to be a type 1 hypervisor this shoud work and one will get a secured embedded platform.
+This should be tested using a Lenovo X220
```
Comment (junjiemao1) [inline stupid_ideas.txt:13]: @openfnord Thanks for bringing up your ideas!

There was a similar idea previously where [Slim Bootloader](https://github.com/slimbootloader/slimbootloader) (i.e. SBL), which is another BIOS designed to be minimal, is used to boot ACRN. The hypervisor, however, is still loaded by the OS loader of SBL, with Multiboot being the boot protocol between SBL and ACRN. AFAIK in coreboot HOBs are used as the interface to payloads, which may be one of the main differences from the previous approach.

Contributions are always warmly welcomed in ACRN community. Feel free to bring up any draft you have 😄

**Comment by openfnord** [comment]
Hi, there is afik no special interface for the payload in coreboot.
As with the linux kernel there is a binary that is aligned in flash and then gets
executed by coreboot. Coreboot can be built with differet payloads. And no, you do not
need seabios if you dont use bios-routines in acrn.
https://doc.coreboot.org/payloads.html

So I am pretty sure that you can add acrn hypervisor to coreboot.

The Laptop X220 by lenovo is quite well supported by coreboot. As it is a quite
cheap laptop on the 2nd hand market and provides hypervisor support, it
should be good enough to prototype coreboot with acrn.

Who would like to (help) analyzing how to combine coreboot with acrn?

Cheers,

Ludwig

Am Dienstag, September 13, 2022 09:12 CEST, schrieb Junjie Mao ***@***.***>:
 
 
@junjiemao1 commented on this pull request.______________________________________________________________________________

In stupid_ideas.txt:> @@ -0,0 +1,18 @@
+This file just holds my stupid ideas.
+
+Stupid ideas are stupid, non verified ideas, I just got while reading something here and there and getting some thoughts.
+Maybe this is helpful for me or others some point in time later.
+
+ACRN-Hypervisor:
+ACRN is booted using EFI and GRUB, this is complicated.
+So as ACRN wants to be embedded, so combine it with coreboot and boot into the hypervisor right away.
+Having a running ACRN hypervisor just after board bring up, booted from the flash of the x86_64 mainboard's 
+BIOS flash gives you a chain of trust while booting. 
+In the hypervisor ACRN you can try to boot something like linux, xBSD or commecrial operating systems.
+As ACRN claims to be a type 1 hypervisor this shoud work and one will get a secured embedded platform.
+This should be tested using a Lenovo X220 

@openfnord Thanks for bringing up your ideas!
There was a similar idea previously where Slim Bootloader (i.e. SBL), which is another BIOS designed to be minimal, is used to boot ACRN. The hypervisor, however, is still loaded by the OS loader of SBL, with Multiboot being the boot protocol between SBL and ACRN. AFAIK in coreboot HOBs are used as the interface to payloads, which may be one of the main differences from the previous approach.
Contributions are always warmly welcomed in ACRN community. Feel free to bring up any draft you have 😄

**Comment by junjiemao1** [comment]
Let's see who volunteers to lead. We are all here to provide inputs.

**Comment by NanlinXie** [comment]
@openfnord , I created github issue to track your question, and close this PR to avoid confusion, please be free to let me know if you have any concern.
https://github.com/projectacrn/acrn-hypervisor/issues/8258

**Pull request closed without merging**